src/bridge/cockpitcpusamples.c

```diff
@@ -112,3 +112,81 @@ cockpit_cpu_samples (CockpitSamples *samples)
   g_strfreev (lines);
   g_free (contents);
 }
+
+gchar*
+read_hwmon_file (const gchar *path)
+{
+  g_autoptr(GError) error = NULL;
+  gchar *content = NULL;
+
+  if (!g_file_get_contents (path, &content, NULL, &error))
+    {
+      // ENOENT is used to break loops, do not log it
+      if (!g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_NOENT))
+          g_warning ("error reading file: %s", error->message);
+
+      return NULL;
+    }
+
+  return content;
+}
+
+void
+sample_cpu_sensors (CockpitSamples *samples,
+                    int hwmonID)
+{
+  for (int i = 1; TRUE; i++)
+    {
+      g_autofree gchar *temp = g_strdup_printf ("/sys/class/hwmon/hwmon%d/temp%d_input", hwmonID, i);
+      g_autofree gchar *temp_content = read_hwmon_file (temp);
+
+      // end loop on error
+      if (temp_content == NULL)
+          break;
+
+      g_autofree gchar *label = g_strdup_printf ("/sys/class/hwmon/hwmon%d/temp%d_label", hwmonID, i);
+      g_autofree gchar *label_content = read_hwmon_file (label);
+
+      if (label_content == NULL)
+        {
+          // labels aren't used on ARM
+          label_content = g_strdup_printf ("Core %d", i);
+        }
+
+      g_strchomp (label_content);
+
+      // ignore Tctl on AMD devices
+      if (g_str_equal (label_content, "Tctl"))
+          continue;
+
+      gint temperature = g_ascii_strtoll (temp_content, NULL, 10);
```

Comment:
This can fail on invalid strings. See the [docs](https://docs.gtk.org/glib/func.ascii_strtoll.html). Not crucial, but passing "0 degress" forever when the file is e.g. empty or contains some error message instead isn't nice.

Please add an `endptr`, validate it, and check `errno` on a zero return value. You can also keep it simple and just generally ignore "0" values -- it's relatively safe to say that zero Celsius is rather unrealistic, unless we are talking space telescopes or mars rovers :grin:

Reply:
I added some error handling.